api/src/main/java/org/apache/iceberg/catalog/SessionCatalog.java

```diff
@@ -147,10 +147,11 @@ public Map<String, String> properties() {
    * @param context session context
    * @param ident a table identifier
    * @param metadataFileLocation the location of a metadata file
+   * @param force default is false. If true, do register table even if table exists otherwise it will throw error.
    * @return a Table instance
    * @throws AlreadyExistsException if the table already exists in the catalog.
    */
-  Table registerTable(SessionContext context, TableIdentifier ident, String metadataFileLocation);
+  Table registerTable(SessionContext context, TableIdentifier ident, String metadataFileLocation, boolean force);
```

Comment:
It is a bit confusing for user by reusing the name `registerTable` for updating the metadata location, even with the `force` option. Also `updating metadata location` is a risky operation, we need to separate it from a safe operation(register a table), so that user won't accidentally do that. Suggested names would be `resetTable` or `resetMetadataLocation`.

Reply:
I think a new procedure will be better. Would like to hear from others. @szehon-ho @RussellSpitzer.

Reply:
I liked it being just an additional parameter, but I think I'm outvoted here :)

Reply:
Based on @szehon-ho 's suggestion below perhaps this should be "resetTable" based on the `forReset` comments below

Reply:
ok